packages/custom_lint/lib/src/workspace.dart

```diff
@@ -602,21 +602,28 @@ publish_to: 'none'
   }
 
   void _writePubspecDependencies(StringBuffer buffer) {
+    // Collect all the dependencies for each package.
     final uniqueDependencyNames = projects.expand((e) sync* {
+      yield* e.pubspec.dependencies.keys;
       yield* e.pubspec.devDependencies.keys;
       yield* e.pubspec.dependencyOverrides.keys;
     }).toSet();
 
     final dependenciesByName = {
       for (final name in uniqueDependencyNames)
         name: (
-          devDependencies: projects
+          dependencies: projects
               .map((project) {
-                final dependency = project.pubspec.devDependencies[name];
-                if (dependency == null) return null;
-                return (project: project, dependency: dependency);
+                final dependency = project.pubspec.dependencies[name];
+                final devDependency = project.pubspec.devDependencies[name];
+                return [
+                  if (dependency != null)
+                    (project: project, dependency: dependency),
+                  if (devDependency != null)
+                    (project: project, dependency: devDependency),
+                ];
               })
-              .nonNulls
+              .expand((e) => e)
```

Comment:
Nit but we should be able to do `.expand(...)` directly instead of `.map().expand()`